ext/storage_sources/s3_store/CMakeLists.txt

```diff
@@ -0,0 +1,82 @@
+project(s3_store CXX)    
+    
+include(ExternalProject)
+include(GNUInstallDirs)
+
+if (ENABLE_S3)
+    # Download and install the AWS CPP SDK into the build directory.
+    ExternalProject_Add(aws-sdk
+        PREFIX aws-sdk-cpp
+        GIT_REPOSITORY    https://github.com/aws/aws-sdk-cpp.git
+        GIT_TAG           1.9.175
+        LIST_SEPARATOR    "|"
+        CMAKE_ARGS       -DBUILD_SHARED_LIBS=ON
+                        -DBUILD_ONLY=s3-crt
+                        -DCMAKE_INSTALL_PREFIX=${CMAKE_CURRENT_BINARY_DIR}/aws-sdk-cpp/install
+                        -DCMAKE_POSITION_INDEPENDENT_CODE=ON
+        BUILD_ALWAYS    FALSE
+        INSTALL_DIR     ${CMAKE_CURRENT_BINARY_DIR}/aws-sdk-cpp/install
+        BUILD_BYPRODUCTS
+        ${CMAKE_CURRENT_BINARY_DIR}/aws-sdk-cpp/install/${CMAKE_INSTALL_LIBDIR}/libaws-cpp-sdk-s3-crt${CMAKE_SHARED_LIBRARY_SUFFIX}
+        ${CMAKE_CURRENT_BINARY_DIR}/aws-sdk-cpp/install/${CMAKE_INSTALL_LIBDIR}/libaws-cpp-sdk-core${CMAKE_SHARED_LIBRARY_SUFFIX}
+        TEST_COMMAND      ""
+        UPDATE_COMMAND    ""
+    )
+    ExternalProject_Get_Property(aws-sdk INSTALL_DIR)
+
+    add_library(aws-sdk::core SHARED IMPORTED)
+    add_library(aws-sdk::s3-crt SHARED IMPORTED)
+    add_library(aws-sdk::crt SHARED IMPORTED)
+    set_target_properties(aws-sdk::core PROPERTIES
+        IMPORTED_LOCATION ${INSTALL_DIR}/${CMAKE_INSTALL_LIBDIR}/libaws-cpp-sdk-core${CMAKE_SHARED_LIBRARY_SUFFIX}                                             
+    )
+    set_target_properties(aws-sdk::s3-crt PROPERTIES
+        IMPORTED_LOCATION ${INSTALL_DIR}/${CMAKE_INSTALL_LIBDIR}/libaws-cpp-sdk-s3-crt${CMAKE_SHARED_LIBRARY_SUFFIX})
+    add_dependencies(aws-sdk::core aws-sdk)
+    add_dependencies(aws-sdk::s3-crt aws-sdk)
+
```

Comment:
Everything above this line should probably exist in a separate file. For managing third party build dependencies (such as the AWS SDK), we store in files under `cmake/third_party`. So the above could be in `cmake/third_party/aws_sdk.cmake`